File: day01/part1.py
#Read data 
text_file = open("input.txt", "r")
lines = [int(v.strip("\n")) for v in text_file.readlines()]

acc=0
for i,num in enumerate(lines): #iterate over each number
    print(i,num)
    if i==0: #skip the first number b/c you cant compare it to the prior number
        continue
    else:
        if lines[i]>lines[i-1]: #if the number is more than the prior number then add 1
            acc+=1
print("Final Answer:",acc)

File: day01/lua/part1.lua
-- function to read data
-- stolen from here: https://stackoverflow.com/questions/11201262/how-to-read-data-from-a-file-in-lua
function lines_from(file)
    lines = {}
    for line in io.lines(file) do 
      lines[#lines + 1] = tonumber(line)
    end
    return lines
end

-- Read the data
  local file = '../input.txt'
  local lines = lines_from(file)

-- Function to convert bool to number
-- stolen from here: https://stackoverflow.com/questions/48230472/boolean-to-number-in-lua
  function bool_to_number(value)
    return value and 1 or 0
  end

  acc=0
  for k,v in pairs(lines) do
    if k ~= 1 then
        acc=acc+bool_to_number(lines[k]>lines[k-1])
    end
    print(k,v,acc)

  end

  print("Final Answer",acc)